### PR TITLE
docs: Introduce the k8s compatibility matrix for KEDA

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

### CODE REVIEW

File: content/docs/2.9/operate/cluster.md [hunk 1]
```diff
@@ -15,6 +15,14 @@ However, maintainers can decide to extend this by supporting more minor versions
 > Example - At time of writing, Kubernetes 1.25 is the latest minor version so KEDA can only use new features that were introduced in 1.23
```
Comment (tomkerkhove) [inline content/docs/2.9/operate/cluster.md:15]: I think we can drop this note now, what do you think?

File: content/docs/2.9/operate/cluster.md [hunk 2]
```diff
@@ -15,6 +15,14 @@ However, maintainers can decide to extend this by supporting more minor versions
...
 You can learn more about the currently supported Kubernetes version in our [FAQ](https://keda.sh/docs/latest/faq/).
```
Comment (tomkerkhove) [inline content/docs/2.9/operate/cluster.md:16]: I think we should remove this entry as well. The FAQ is not version (see https://github.com/kedacore/keda-docs/issues/936) so I think we should point from the FAQ entry to this page but can be done seperately

File: content/docs/2.9/operate/cluster.md [hunk 3]
```diff
@@ -15,6 +15,14 @@ However, maintainers can decide to extend this by supporting more minor versions
...
 
+As a reference, this compatibility matrix shows supported k8s versions per KEDA version:
+
+|   KEDA    |   Kubernetes  |
+|-----------|---------------|
+|   v2.9.x  |   v1.23-v1.25 |
+|   v2.8.x  |   v1.17-v1.24 |
+|   v2.7.x  |   v1.17-v1.24 |
```
Comment (tomkerkhove) [inline content/docs/2.9/operate/cluster.md:24]: Nits for readability, not a must:
```suggestion
|   v2.9  |   v1.23 - v1.25 |
|   v2.8  |   v1.17 - v1.24 |
|   v2.7  |   v1.17 - v1.24 |
```
Let's also be consistent on versions and only show major.minor similar to Kubernetes version

**Review by tomkerkhove — APPROVED** [review]
LGTM, thanks!

**Comment by JorTurFer** [comment]
I'll apply these changes asap, but I have to travel till next week

**Pull request merged**